File: components/ntp_snippets/remote/test_utils.cc
// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/ntp_snippets/remote/test_utils.h"

#include <memory>

#include "components/prefs/testing_pref_service.h"
#include "components/sync/driver/fake_sync_service.h"

namespace ntp_snippets {

namespace test {

FakeSyncService::FakeSyncService()
    : can_sync_start_(true),
      is_sync_active_(true),
      configuration_done_(true),
      is_encrypt_everything_enabled_(false),
      active_data_types_(syncer::HISTORY_DELETE_DIRECTIVES) {}

FakeSyncService::~FakeSyncService() = default;

bool FakeSyncService::CanSyncStart() const {
  return can_sync_start_;
}

bool FakeSyncService::IsSyncActive() const {
  return is_sync_active_;
}

bool FakeSyncService::ConfigurationDone() const {
  return configuration_done_;
}

bool FakeSyncService::IsEncryptEverythingEnabled() const {
  return is_encrypt_everything_enabled_;
}

syncer::ModelTypeSet FakeSyncService::GetActiveDataTypes() const {
  return active_data_types_;
}

RemoteSuggestionsTestUtils::RemoteSuggestionsTestUtils()
    : pref_service_(std::make_unique<TestingPrefServiceSyncable>()) {
  fake_sync_service_ = std::make_unique<FakeSyncService>();
}

RemoteSuggestionsTestUtils::~RemoteSuggestionsTestUtils() = default;

}  // namespace test

}  // namespace ntp_snippets

File: tools/cygprofile/test_utils.py
# Copyright 2017 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Test utilities for cygprofile scripts."""

import collections

import process_profiles

SimpleTestSymbol = collections.namedtuple(
    'SimpleTestSymbol', ['name', 'offset', 'size'])


class TestSymbolOffsetProcessor(process_profiles.SymbolOffsetProcessor):
  def __init__(self, symbol_infos):
    super(TestSymbolOffsetProcessor, self).__init__(None)
    self._symbol_infos = symbol_infos


class TestProfileManager(process_profiles.ProfileManager):
  def __init__(self, filecontents_mapping):
    super(TestProfileManager, self).__init__(filecontents_mapping.keys())
    self._filecontents_mapping = filecontents_mapping

  def _ReadOffsets(self, filename):
    return self._filecontents_mapping[filename]
